packages/did-comm/src/didcomm.ts

```diff
@@ -494,15 +496,60 @@ export class DIDComm implements IAgentPlugin {
     return { msgObj, mediaType }
   }
 
-  private findPreferredDIDCommService(services: any) {
+  private findPreferredDIDCommService(services: Service[]) {
     // FIXME: TODO: get preferred service endpoint according to configuration; now defaulting to first service
     return services[0]
   }
 
+  private async wrapDIDCommForwardMessage(
+    recipientDidUrl: string,
+    messageId: string,
+    packedMessageToForward: IPackedDIDCommMessage,
+    routingKey: string,
+    context: IAgentContext<IDIDManager & IKeyManager & IResolver>,
+  ): Promise<IPackedDIDCommMessage> {
+    const splitKey = routingKey.split('#')
+    const shouldUseSpecificKid = splitKey.length > 1
+    const mediatorDidUrl = splitKey[0]
+    // 1. Create forward message
+    const forwardMessage: IDIDCommMessage = {
+      id: uuidv4(),
+      type: 'https://didcomm.org/routing/2.0/forward',
+      to: mediatorDidUrl,
+      body: {
+        next: recipientDidUrl,
+      },
+      attachments: [
+        {
+          media_type: DIDCommMessageMediaType.ENCRYPTED,
+          data: {
+            json: JSON.parse(packedMessageToForward.message),
+          },
+        },
+      ],
+    }
+
+    context.agent.emit('DIDCommV2Message-forwarded', {
```

Comment:
we want to emit this message at some point, but probably not here, since this is just the message wrapping. this event shouldn't get emitted until after the message is actually forwarded.

Reply:
What about if the event was `DIDCommV2Message-wrappedForward`? In the case of their being more than one mediator, multiple messages would be wrapped but only one would be forwarded. Should this emit one message per wrap? Or only one message for the single forward?